### DISCUSSION TITLE
Flujo del programa

### DISCUSSION THREAD

**Discussion by ypne** [IIC2413/Syllabus-2024-2#57] [General]
Hola
La tarea tiene 2 fases:
1.- Lectura y preparación de los datos
2.- Ejecución del programa Bananer.php

En la primera  parte deben:
a.- Leer los datos desde el archivo de input, detectanto y si es posible corrigiendo en PHP los datos que no corresponden al formato (por ejemplo caracteres inválidos)
b.- Insertar los datos en las matrices, detectando los datos que no corresponden (por ejemplo, letras en un número telefónico). En  caso de haber este tipo de violación al tipo de dato se debe escribir en un segundo archivo la tupla encontrada.
c.- Modificar manualmente o automáticamente(si es posible) todas las tuplas que no se pudieron cargar.
Repetir este proceso hasta que los datos están limpios.

2.- Ejecución del programa Bananer
Una vez teniendo el input corregido comienza  la fase de ejecutar el programa el cual debe hacer lo siguiente:
a.- Cargar desde el/los archivos de input todos los datos ya limpios
b.- Una vez cargados escribir en pantalla "ingrese RUN" y esperar que se ingrese un número, luego <enter>
     b.1.- si el formato del RUN es inválido(o sea tiene letras, es más largo, etc.) desplegar por pantalla "RUN inválido" y volver al prompt
     b.2.- si el RUN no existe en la base desplegar por pantalla "RUN no tiene datos" y volver al prompt
c.- desplegar en pantalla o en un archivo de salida el resultado
d.- Ahora escribir en pantalla "ingrese curso"  y esperar que se ingrese una sigla, luego <enter>
     d.1.- si el formato de l sigla es inválida(o sea tiene caracteres diferente a numeros y letras A-Z, es más largo, etc.) desplegar por pantalla "SIGLA inválida" y volver al prompt
     d.2.- si la SIGLA no existe en la base desplegar por pantalla "SIGLA no tiene datos" y volver al prompt
e.- desplegar en pantalla o en un archivo de salida el resultado
FIN

**Comment by cataruze** [comment]
Hola, yo tenía entendido que los archivos a leer eran los entregados junto con el enunciado, pero ahora mencionan que será por medio de input, cual de los dos casos se aplica a la tarea?

**Reply by ypne** [reply]
Los archivos a leer son del enunciado, si lo ingresas en duro en el programa no es una buena práctica, es mejor dejarlo como parámetro. Sin embargo, eso no se tomará en cuenta en la evaluación.

**Reply by antoer** [reply]
Y las matrices que cremos debemos mostrarlas en pantalla tambien o solo crealas?

**Reply by ypne** [reply]
el enunciado dice que se deben mostrar los resultados de las consultas

**Comment by Estebvn** [comment]
Hola, cuando dices "Modificar **manualmente** o automáticamente(si es posible) todas las tuplas que no se pudieron cargar", no me queda muy claro a qué se refiere con manualmente.
Gracias.

**Reply by ypne** [reply]
ver [Discussion #86](https://github.com/IIC2413/Syllabus-2024-2/discussions/86)

**Comment by javirossi** [comment]
Hola. El enunciado dice que para la lista de curso se debe entregar "cada estudiante perteneciente a un curso y periodo". ¿Se debe ingresar el periodo también? Ya que en lo que señalas no se pide un periodo.

**Reply by ypne** [reply]
asume el periodo vigente si quieres, otra solucion tambien es válida

**Comment by ssepul21** [comment]
Hola! 
 En la primera parte, ¿se busca que las tuplas que tienen errores vuelvan a ser incorporados en la base de datos "que tiene datos correctos" una vez corregidos?
 No me queda del todo claro el manejo que se espera que hagamos con estos datos. 

Gracias de antemano!

**Reply by ypne** [reply]
ver discussions #86 ([url](https://github.com/IIC2413/Syllabus-2024-2/discussions/86))

**Comment by ypne** [comment]
esto se merece otra discussion [https://github.com/IIC2413/Syllabus-2024-2/discussions/86](url)

**Comment by pedrorojasi** [comment]
Yo lo hice de manera que se ingrese el tipo de información como en un "menú" que ofrece las diversas matrices según el ingreso de números(por ejemplo, si se introduce 1 vas a la interfaz de matriz "Persona"), para luego poner rut o sigla de curso dependiendo de  que y de donde se quiere conseguir la información.
Se muestra todo lo pedido por la misma consola
¿Eso estaria bien para el flujo de el programa?

**Comment by ypne** [comment]
no me queda muy claro lo que estas haciendo, porqué habría que seleccionar la matriz?

**Reply by pedrorojasi** [reply]
las matrices luego de hacerlas las debo cargar en archivo o puedo entregarlas por consola directamente? Porque estaba entregando todo por consola. Es por eso que existe esa selección, para comprobar que se entreguen las respectivas matrices.
Y solo imprima la que se quiera revisar, no cualquiera.

**Reply by ypne** [reply]
De enunciado:
"• Realizar las siguientes consultas sobre los datos, entregando el resultado por consola o en archivo:
1. Carga acad ́emica acumulada: Entregar, en una l ́ınea, cada curso realizado (cualquiera sea la calificaci ́on) por un estudiante identificado por su RUN ingresado por consola, indicando: periodo del curso, sigla del curso, curso, nota y calificaci ́on. Al final de cada per ́ıodo la nota promedio del semestre (PPS) y al final del reporte la nota promedio de todos los cursos (PPA)
2. Lista de curso: Entregar, en una l ́ınea, cada estudiante perteneciente a un curso y periodo, especificado por consola, indicando: cohorte, nombre completo, RUN y nu ́mero de estudiante"
